configure-clusters/src/main/java/com/esri/arcgismaps/sample/configureclusters/screens/MainScreen.kt

```diff
@@ -0,0 +1,398 @@
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SheetState
+import androidx.compose.material3.Switch
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.esri.arcgismaps.sample.configureclusters.components.MapViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.BottomSheet
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import com.esri.arcgismaps.sample.sampleslib.theme.SampleTypography
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlin.math.roundToInt
+
+/**
+ * Main screen layout for the sample app
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MainScreen(sampleName: String) {
+    // create a ViewModel to handle MapView interactions
+    val mapViewModel: MapViewModel = viewModel()
+
+    val composableScope = rememberCoroutineScope()
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+                contentAlignment = Alignment.Center
+            ) {
+                var mapScale by remember { mutableIntStateOf(0) }
+                MapView(
+                    modifier = Modifier
+                        .fillMaxSize(),
+                    mapViewProxy = mapViewModel.mapViewProxy,
+                    // identify on single tap
+                    onSingleTapConfirmed = { singleTapConfirmedEvent ->
+                        mapViewModel.identify(singleTapConfirmedEvent)
+                    },
+                    arcGISMap = mapViewModel.arcGISMap,
+                    // update the map scale in the UI on map scale change
+                    onMapScaleChanged = { currentMapScale ->
+                        if (!currentMapScale.isNaN()) {
+                            mapScale = currentMapScale.roundToInt()
+                        }
+                    },
+                )
+
+                val controlsBottomSheetState = rememberModalBottomSheetState()
```

Comment:
Let's use this flag to fully expand the sheet when `.show()` is called. Alternatively you could use, `.expand()` which will also skip the partially expanded state.  
```suggestion
                val controlsBottomSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
```

Reply:
This behaviour is much better. I couldn't figure out why it wasn't closing reliably. Thanks!